### ISSUE TITLE
Publish Build Info Task fails (using projectKey)

### ISSUE THREAD

**Issue by Tobias-08** [jfrog/artifactory-azure-devops-extension#269]
**Describe the bug**
- Publish Build Info task fails when projectKey is specified
- In contrast publish succeeds when using the global build repository (no projectKey)
- Log extract:

```
[Error] open C:\Users\[...]\AppData\Local\Temp\jfrog\builds\RGF0YXBvcnROU2VydmljZUJ1c0VuZHBvaW50XzEuMC4yM19kbDQzd3A=\partials\details: The system cannot find the file specified.
##[error]Error: Command failed: E:\BA\A6\_work\_tool\jfrog\1.51.1\x64\jfrog.exe rt bp "[...]" "1.0.23" --url="https://[...]/artifactory" --env-exclude="*password*;*secret*;*key*;*token*" --build-url="https://[...]/_build?buildId=61110" --user=*** --password=*** --project="[...]"

```

**To Reproduce**
- Empty project-level build repository
- use publish build info task with projectKey

**Expected behavior**
- Publish Build Info with projectKey works

**Versions**
- Azure Devops: 2020 (on premise)
- Artifactory Azure DevOps extension version: 1.13.0
- jfrog CLI version: 1.51.1 (but same behavior with latest 2.x)
- Agent operating system: Windows
- JFrog Artifactory version: 7.24.3

**Comment by eyalbe4** [comment]
@Tobias-08,
Do you have an Artifactory task configured, which is running prior to the **Artifactory Publish Build-Info** task, and has the the project key configured? If you don't, then I suspect this can be the cause of this error.
This is because the task fails when it tries to find the directory on the local file system, which includes the build-info details collected by the previous tasks. The directory name is a base 64 representation of the project key, build name and build number. If therefore the project key hasn't been used by previous tasks to create this directory, the publish build-info task will fail when trying to find it. We should definitely catch this and prompt a better error message. 
Please let us know however if the above is indeed the root cause and if this fixes the issue for you.

**Comment by Tobias-08** [comment]
@eyalbe4 
Yes, the Artifactory restore task was missing the projectKey. Adding it solved the issue, thanks!

A better error message would be helpful, though.
